### PR TITLE
Visualization rework

### PR DESCRIPTION
**Pull request by asalzburger**
This PR harmonies the visualisation to a large extend.

Except for `BoundingBox`, `Ray` and `Frustum` none of the Acts classes have now `draw()` methods anymore. Those will follow with #269.

It clearly separates the drawing visitor classes (`IVisualisation` base class, `PlyVisualization` and `ObjVisualization` specialisations) from the helper methods to use these  in helper structs:
* GeometryView
* EventDataView

Both of these files are now compilation units rather than fully inlined.

It introduces a `ViewConfig` struct that allows custom setting of colours and other parameters without extensive interfaces.

It also introduces the drawing of `TrackingVolume` and `Layer` objects and thus removes the duplicated infrastructure of obj writing from `Examples/Io/Obj` which duplicated a lot of visualisation code.

Finally it comes with a new `visualisation` documentation with some sample screenshots.

There are still some issues with the ply visitor, which are followed up in: #271

### CODE REVIEW

**Comment by asalzburger** [comment]
Hi @Corentin-Allaire - since you mentioned that there are many `.obj` files in the tests.

This PR removes several of them, and - as the description says - will be followed by others that remove a view more files being written.

It does a first cleanup of the visualisation (as described).

P.S. The new view grid technology, could probably also be used to display material maps on layers, but I guess that's something for later.

**Comment by asalzburger** [comment]
surfaces-and-grid00

**Comment by asalzburger** [comment]
ping @Corentin-Allaire

File: Tests/UnitTests/Core/Visualization/TrackingGeometryViewTests.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  auto plyErrors3M = testPlyString(plyTest3M, triangulate);
+  std::cout << "Sufaces Ply Test 3M : " << plyTest3M.size()
+            << " characters written with " << plyErrors3M.size() << " errors."
+            << std::endl;
+  BOOST_CHECK(plyErrors3M.size() == 0);
+  for (auto plyerr : plyErrors3M) {
+    std::cout << plyerr << std::endl;
+  }
+}
+*/
```
Comment (Corentin-Allaire) [inline Tests/UnitTests/Core/Visualization/TrackingGeometryViewTests.cpp:80]: Sorry if I missed it but why is the test for the ply output  commented ?

File: Core/include/Acts/Visualization/EventDataView.hpp [hunk 2]
```diff
@@ -0,0 +1,272 @@
+// This file is part of the Acts project.
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/include/Acts/Visualization/EventDataView.hpp:1]: The renaming is there to separate `Visualisation` which are visitors that implement `IVisualizaiton` interface and `View` which use them.

File: Core/include/Acts/Visualization/ViewConfig.hpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  bool visible = true;
+  /// The RGB color for this object
+  ColorRGB color = {250, 0, 0};
+  /// Out of plane drawing parameter for objects
+  double offset = 0.1;
+  /// The visual line thickness for this object
+  double lineThickness = 0.15;
+  /// The visual surface thickness for this object
+  double surfaceThickness = 0.15;
+  /// The number of segmetns to approximate full 2pi
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/include/Acts/Visualization/ViewConfig.hpp:39]: Fixed.

File: Core/src/Visualization/GeometryView.cpp [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+
+void Acts::GeometryView::drawTrackingVolume(
+    IVisualization& helper, const TrackingVolume& tVolume,
+    const GeometryContext& gctx, const ViewConfig& containerView,
+    const ViewConfig& volumeView, const ViewConfig& layerView,
+    const ViewConfig& sensitiveView, const ViewConfig& gridView, bool writeIt,
+    const std::string& tag) {
+  if (tVolume.confinedVolumes() != nullptr) {
+    const auto& subVolumes = tVolume.confinedVolumes()->arrayObjects();
+    for (const auto& tv : subVolumes) {
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/src/Visualization/GeometryView.cpp:186]: I will update, once we have a geometry to test.

**Comment by asalzburger** [comment]
Hey @Corentin-Allaire -  I addressed the comments, I suppose - I also integrated a type problem which Tulay pointed out.

**Comment by Corentin-Allaire** [comment]
Did I miss the answer on the  ply output for the TrackingGeometryTest ?
Except from that I can approve.

**Comment by asalzburger** [comment]
> Did I miss the answer on the ply output for the TrackingGeometryTest ?
> Except from that I can approve.

This is explained in the description of the PR:

There are still some issues with the ply visitor, which are followed up in: #271

(I don't want to do it here, as the MR is already big).

**Comment by Corentin-Allaire** [comment]
> > Did I miss the answer on the ply output for the TrackingGeometryTest ?
> > Except from that I can approve.
> 
> This is explained in the description of the PR:
> 
> There are still some issues with the ply visitor, which are followed up in: #271
> 
> (I don't want to do it here, as the MR is already big).

Ah yes, I missed that sorry...

**Pull request merged**